src/NuGet.Clients/VsExtension/VsCredentialProviderImporter.cs

```diff
@@ -75,36 +76,40 @@ public class VsCredentialProviderImporter
             _initializer = initializer ?? Initialize;
         }
 
-        [Import("VisualStudioAccountProvider", typeof(IVsCredentialProvider), AllowDefault = true)]
-        public IVsCredentialProvider ImportedProvider { get; set; }
+        [ImportMany(typeof(IVsCredentialProvider))]
+        public IEnumerable<IVsCredentialProvider> ImportedProviders { get; set; }
 
         /// <summary>
         /// Plugin providers are entered loaded the same way as other nuget extensions,
         /// matching any extension named CredentialProvider.*.exe.
         /// </summary>
         /// <returns>An enumeration of plugin providers</returns>
-        public ICredentialProvider GetProvider()
+        public IReadOnlyCollection<ICredentialProvider> GetProviders()
         {
-            ICredentialProvider result = null;
+            var results = new List<ICredentialProvider>();
 
             try
             {
                 _initializer();
 
-                if (ImportedProvider != null)
+                if (ImportedProviders != null)
                 {
-                    result = new VsCredentialProviderAdapter(ImportedProvider);
+                    // Ensure imported providers ordering is deterministic
+                    foreach (var importedProvider in ImportedProviders.OrderBy(p => p.GetType().Name))
+                    {
+                        results.Add(new VsCredentialProviderAdapter(importedProvider));
+                    }
                 }
 
                 // Dev15+ will provide a credential provider for VSTS.
                 // If we are in Dev14, and no imported VSTS provider is found,
                 // then fallback on the built-in VisualStudioAccountProvider
-                if (result == null && IsDev14)
+                if (results.Count == 0 && IsDev14)
```

Comment:
The check needs to be stronger here: we need to make sure that we have loaded the VisualStudioAccountProvider, and this can be done either by contract name or by full type name.

The issue here is that we use the same version of NuGet VSIX in dev14 and dev15, dev15 can load the VisualStudioAccountProvider via MEF and dev14 cannot, and dev14 can use the built in version of this provider and dev15 cannot due to dependencies.